### PR TITLE
Use cross platform config directory

### PR DESCRIPTION
**Pull request by tcharding**
Currently we hard code the `.config/` path.  We can be cross platform if we use the `directories` crate instead.

Implements solution to #1074 for btsieve.  cnd coming as a separate PR.

### CODE REVIEW

File: btsieve/src/load_settings.rs [hunk 1]
```diff
@@ -12,34 +13,30 @@ pub struct Opt {
...
-        Some(config_file) => {
-            if config_file.exists() {
-                Settings::read(config_file)
-            } else {
-                Err(ConfigError::Message(format!(
-                    "Could not load config file: {:?}",
-                    config_file
-                )))
+        Some(file) => parse_config_file(file),
+        None => match ProjectDirs::from("tech", "CoBloX", "btsieve") {
```
Comment (thomaseizinger) [inline btsieve/src/load_settings.rs]: Do we want this to say comit-network instead?
```suggestion
        None => match ProjectDirs::from("network", "comit, "btsieve") {
```

Comment (tcharding) [reply btsieve/src/load_settings.rs]: Cool, I like it.

Comment (bonomat) [reply btsieve/src/load_settings.rs]: Please also update the comments below.

Comment (bonomat) [reply btsieve/src/load_settings.rs]: Without `network` (or `tech`) the path resolves to: 
`"/Users/<username>/Library/Preferences/comit-network.btsieve/btsieve.toml"
`

Comment (tcharding) [reply btsieve/src/load_settings.rs]: > Please also update the comments below.

I don't understand what you mean @bonomat, can you explain please (if action is still required), thanks.

**Pull request merged**